### PR TITLE
Wizard recipe: ORC-v4.31.0

### PR DESCRIPTION
**Pull request by bramtayl**
This pull request contains a new build recipe I built using the BinaryBuilder.jl wizard:

* Package name: ORC
* Version: v4.31.0

@staticfloat please review and merge.

### CODE REVIEW

**Comment by staticfloat** [comment]
@bramtayl just in case you don't know, you can edit the pull request here in the web browser and add commits to the PR, and Yggdrasil will re-run CI.  So you don't have to keep on re-running the wizard and opening new PRs, you can just manually tweak the script that is generated from your wizard session.  I'm glad you apparently like the wizard so much, but I want to save you as much work as possible.  :)

**Comment by bramtayl** [comment]
Thanks! The wizard is lovely. I'm not quite sure what's going on here, but I'm working on it.

**Comment by bramtayl** [comment]
Well, I'm stuck

File: O/ORC/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+exit
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = supported_platforms()
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct("liborc-test", :liborc_test),
```
Comment (giordano) [inline O/ORC/build_tarballs.jl:30]:
```suggestion
    LibraryProduct("liborc-test-0.4", :liborc_test),
```

Comment (bramtayl) [reply O/ORC/build_tarballs.jl:30]: Oh got it. I think these statements were generated by the wizard though, so maybe it made a mistake?

Comment (giordano) [reply O/ORC/build_tarballs.jl:30]: But you inserted the name

Comment (bramtayl) [reply O/ORC/build_tarballs.jl:30]: Well, I inserted the names with underscores, but not the ones with dashes, I don't think? I pasted the products part of the wizard below

```
[press: d=done, a=all, n=none]
 > [X] bin/orc-bugreport
   [X] bin/orcc
   [X] lib/liborc-0.4.so.0.31.0
   [X] lib/liborc-test-0.4.so.0.31.0
Please provide a unique variable name for each build artifact:
lib/liborc-test-0.4.so.0.31.0:
> liborc_test

bin/orcc (default 'orcc'):
> 

lib/liborc-0.4.so.0.31.0:
> liborc

bin/orc-bugreport:
> orc_bugreport
```

File: O/ORC/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = supported_platforms()
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct("liborc-test-0.4", :liborc_test),
+    ExecutableProduct("orcc", :orcc),
+    LibraryProduct("liborc", :liborc),
```
Comment (giordano) [inline O/ORC/build_tarballs.jl:32]:
```suggestion
    LibraryProduct("liborc-0.4", :liborc),
```

File: O/ORC/build_tarballs.jl (hunk 1 above)
Comment (giordano) [reply O/ORC/build_tarballs.jl]: Right, sorry, this is a bug in the wizard, which happens when libraries have stupid names, like that.  We've had in another occasion too

**Pull request merged**